common/src/main/java/org/mcaccess/minecraftaccess/features/point_of_interest/POIEntities.java

```diff
@@ -38,100 +36,104 @@ public class POIEntities {
-                    entity -> entity instanceof ItemEntity itemEntity && itemEntity.isOnGround() || entity instanceof PersistentProjectileEntity projectile && projectile.pickupType.equals(PersistentProjectileEntity.PickupPermission.ALLOWED), null)
+    private static final POIEntities INSTANCE = new POIEntities();
+    private @Nullable Class<? extends Entity> marked = null;
+
+    public final POIGroup<Entity> hostileGroup = new POIGroup<>(
+            SoundEvents.BLOCK_NOTE_BLOCK_BELL.value(),
+            2f,
+            entity -> entity instanceof HostileEntity || entity instanceof Angerable monster && (monster.hasAngerTime() || MinecraftClient.getInstance().player.getUuid().equals(monster.getAngryAt()) || MinecraftClient.getInstance().player.getUuid().equals(monster.getAttacker()))
     );
 
-    static {
-        instance = new POIEntities();
-    }
+    @SuppressWarnings("unchecked")
+    final POIGroup<Entity>[] groups = new POIGroup[] {
+            new POIGroup<Entity>(// Your Pets
+                    SoundEvents.BLOCK_NOTE_BLOCK_FLUTE.value(),
+                    1f,
+                    entity -> entity instanceof TameableEntity pet && MinecraftClient.getInstance().player.getUuid().equals(pet.getOwnerUuid())
+            ),
+            new POIGroup<Entity>(// Other Pets
+                    SoundEvents.BLOCK_NOTE_BLOCK_COW_BELL.value(),
+                    1f,
+                    entity -> entity instanceof TameableEntity pet && pet.isTamed()
+            ),
+            new POIGroup<Entity>(// Bosses
+                    SoundEvents.BLOCK_NOTE_BLOCK_PLING.value(),
+                    2f,
+                    entity -> entity instanceof MobEntity mob && mob.getMaxHealth() >= 80 && !(entity instanceof IronGolemEntity)
+            ),
+            hostileGroup,
+            new POIGroup<Entity>(// Passive Mobs
+                    SoundEvents.BLOCK_NOTE_BLOCK_BELL.value(),
+                    0f,
+                    entity -> entity instanceof PassiveEntity || entity instanceof WaterCreatureEntity
+            ),
+            new POIGroup<Entity>(// Players
+                    SoundEvents.BLOCK_NOTE_BLOCK_CHIME.value(),
+                    1f,
+                    entity -> entity instanceof PlayerEntity
+            ),
+            new POIGroup<Entity>(// Vehicles
+                    SoundEvents.BLOCK_NOTE_BLOCK_IRON_XYLOPHONE.value(),
+                    1f,
+                    entity -> entity instanceof VehicleEntity
+            ),
+            new POIGroup<Entity>(// Items
+                    SoundEvents.BLOCK_METAL_PRESSURE_PLATE_CLICK_ON,
+                    2f,
+                    entity -> entity instanceof ItemEntity itemEntity && itemEntity.isOnGround() || entity instanceof PersistentProjectileEntity projectile && projectile.pickupType.equals(PersistentProjectileEntity.PickupPermission.ALLOWED)
+            ),
+    };
```

Comment:
Same as POI Block, consider extract the groups.

Reply:
That would make it significantly harder to maintain because then some changes such as adding or removing a group will need to be made in multiple places. Order is also significant here so it would be less readable because you would have to reference the order in the array which is seperate at the same time. Is the comment there really that insufficient?